core/src/main/resources/hudson/model/Job/_api.jelly

```diff
@@ -27,6 +27,14 @@ THE SOFTWARE.
 
   <st:include page="/hudson/model/AbstractItem/_api.jelly"/>
 
+  <h2>Retrieving all builds</h2>
+  <p>
+    To prevent Jenkins from having to load all builds from disk when someone accesses the job API, the <code>builds</code>
+    tree only contains the 50 newest builds. If you really need to get all builds, access the <code>allBuilds</code> tree,
```

Comment:
@daniel-beck where did you get 50 from? Even at the time this was 100: https://github.com/daniel-beck/jenkins/blob/ed94d62c342ba9c09e1853f4a0bccf3fee6d9543/core/src/main/java/hudson/model/Job.java#L691-L694